common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -206,8 +207,20 @@ public static ServiceException getExceptionFromTokenErrorResponse(@NonNull final
 
         final ServiceException outErr;
 
-        if (shouldBeConvertedToUiRequiredException(errorResponse.getError())) {
+        if (isNativeAuthenticationMFAError(errorResponse)) {
+            ServiceException apiError = new ServiceException(
+                    errorResponse.getError(),
+                    errorResponse.getErrorDescription(),
+                    null);
 
+            String developerDescription = "Multi-factor authentication is required, which can't be fulfilled as part of this flow. Please sign out and perform a new sign in operation. Please see exception details for more information.";
+            outErr = new ServiceException(
+                    errorResponse.getError(),
+                    developerDescription,
+                    apiError
+            );
+        }
+        else if (shouldBeConvertedToUiRequiredException(errorResponse.getError())) {
```

Comment:
I'm wondering if we should have a dedicated exception handler for Native Auth i.e. NativeAuthExceptionHandler and offload this logic to that class.

Reply:
We're (re-)using the SilentTokenCommand, so I'd have to make changes there too. So given that this is shared logic, I'd prefer to leave it as is.